apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-patterns-plugin.tsx

```diff
@@ -24,13 +25,20 @@ export function PagePatternsPlugin( props: PagePatternsPluginProps ): JSX.Elemen
 	const { disableTips } = useDispatch( 'core/nux' );
 
 	const selectProps = useSelect( ( select ) => {
-		const { isOpen } = select( 'automattic/starter-page-layouts' );
+		const { isOpen, isPatternPicker } = select( 'automattic/starter-page-layouts' );
 		return {
 			isOpen: isOpen(),
 			isWelcomeGuideActive: select( 'core/edit-post' ).isFeatureActive( 'welcomeGuide' ) as boolean, // Gutenberg 7.2.0 or higher
 			areTipsEnabled: select( 'core/nux' )
 				? ( select( 'core/nux' ).areTipsEnabled() as boolean )
 				: false, // Gutenberg 7.1.0 or lower
+			...( isPatternPicker() && {
```

Comment:
What happens when `isPatternPicker` returns false? I think spreading `false` would throw.

Reply:
It's not throwing. Actually, this is a pattern we use quite intensively. See https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/gutenboarding/stores/onboard/actions.ts#L91-L109

Reply:
Yes, we use it for object or undefined|null. But I haven't seen it for booleans.

Reply:
This throws, you can copy it to browser console:

```js
const bool = false;
console.log(...(bool && {name: 'Omar'}))
```

Reply:
It seems even TS isn't happy 
![image](https://user-images.githubusercontent.com/17054134/117990954-e9186880-b33d-11eb-8caa-77f11c0752ad.png)

Reply:
The example is going to throw an error also if `bool` is `undefined`. 
Try wrapping the spread in an object: `console.log( { ...(bool && {name: 'Omar'}) } )`

Reply:
That makes sense! LGTM.

Reply:
I guess in your local environment, TS picks `isPatternPicker()` as falsy. It may be falsy at times but not always.

This may shed some light:

> Spreading in a value that is sometimes a non-object is idiomatic and not a problem.
> Spreading in a value that is always a non-object indicates a logic error in your code.

from: https://github.com/microsoft/TypeScript/issues/37491#issuecomment-601741064

Reply:
Type error went away for me once I restarted the TS server 🤷‍♂️
When I hover over `isPatternPicker` it says `() => boolean`, which is correct and shouldn't cause problems with spreading. Before I restarted the TS server it said something like `() => unknown` which is probably what was causing the problem.